code/modules/vehicles/cars/golfcart/golfcart.dm

```diff
@@ -0,0 +1,677 @@
+	desc = "A small grey key for using the golf cart."
+	icon = 'icons/obj/toys/golfcart_split.dmi'
+
+/obj/item/golfcart_kit
+	name = "golfcart parts kit"
+	desc = "A box containing a golf cart. Some assembly required. Batteries not included."
+	icon = 'icons/obj/toys/golfcart_split.dmi'
+	icon_state = "parts_kit"
+	w_class = WEIGHT_CLASS_HUGE
+	throw_range = 2
+	item_flags = SLOWS_WHILE_IN_HAND | IMMUTABLE_SLOW
+	slowdown = 2.5
+	drag_slowdown = 3.5
+
+/obj/item/golfcart_kit/examine(mob/user)
+	. = ..()
+	. += span_notice("The instructions say that it needs to be [EXAMINE_HINT("screwed")] together.")
+
+/obj/item/golfcart_kit/proc/play_building_noises(mob/living/user, duration)
+	duration = max(duration - (1 SECONDS), 0.5 SECONDS)
+	playsound(src, 'sound/items/poster/poster_ripped.ogg', 50, TRUE)
+	sleep(1 SECONDS)
+	if (!DOING_INTERACTION_WITH_TARGET(user, src))
+		return
+	playsound(src, 'sound/items/tools/screwdriver_operating.ogg', 50, TRUE)
+	sleep(duration / 2)
+	if (!DOING_INTERACTION_WITH_TARGET(user, src))
+		return
+	playsound(src, 'sound/items/tools/ratchet.ogg', 50, TRUE)
+	sleep(duration / 2)
+	if (!DOING_INTERACTION_WITH_TARGET(user, src))
+		return
+	playsound(src, 'sound/items/tools/screwdriver.ogg', 50, TRUE)
+
+/obj/item/golfcart_kit/screwdriver_act(mob/living/user, obj/item/tool)
+	if (!isturf(loc))
+		user.balloon_alert(user, "set down first!")
+		return ITEM_INTERACT_BLOCKING
+	user.visible_message(span_notice("[user] starts putting together the [src]..."), span_notice("You start assembling the [src]..."))
+	var/unboxing_duration = 7 SECONDS
+	INVOKE_ASYNC(src, PROC_REF(play_building_noises), user, unboxing_duration * tool.toolspeed)
+	if(!tool.use_tool(src, user, unboxing_duration))
+		return ITEM_INTERACT_BLOCKING
+	if (!isturf(loc))
+		return ITEM_INTERACT_BLOCKING
+	var/obj/vehicle/ridden/golfcart/cart = new(get_turf(src))
+	user.visible_message(span_notice("[user] assembles the [cart]!"), span_notice("You assemble the [cart]."))
+	qdel(src)
+
+/obj/vehicle/ridden/golfcart/atom_break()
+	. = ..()
+	if (smoke)
+		return
+	smoke = new(src, /particles/smoke/ash)
+
+/obj/vehicle/ridden/golfcart/atom_fix()
+	. = ..()
+	if (smoke)
+		QDEL_NULL(smoke)
+
+/obj/vehicle/ridden/golfcart/atom_destruction()
+	explosion(src, devastation_range = -1, light_impact_range = 2, flame_range = 3, flash_range = 4)
+	return ..()
+
+///Jiggles the cargo_image up and down.
+/obj/vehicle/ridden/golfcart/proc/shake_cargo(pixelshiftx = 2, pixelshifty = 2, duration)
+	if (!cargo_image)
+		return
+	var/inital_pixel_x = cargo_image.pixel_x
+	var/inital_pixel_y = cargo_image.pixel_y
+	animate(cargo_image, pixel_x = inital_pixel_x + rand(-pixelshiftx, pixelshiftx), pixel_y = inital_pixel_y + rand(pixelshifty/2, pixelshifty), time=duration, flags=ANIMATION_PARALLEL)
+	animate(pixel_x = inital_pixel_x, pixel_y = inital_pixel_y, time=duration)
+
+/obj/vehicle/ridden/golfcart/proc/is_hotrod()
+	return engine && engine_state && engine_state == ENGINE_WELDED
```

Comment:
```suggestion
	return engine && engine_state == ENGINE_WELDED
```
There's no situation where you care about the engine being in the car if it's not welded in, that you check is_hotrod()